src/BunqLib.php

```diff
@@ -437,36 +437,65 @@ public function addCallbackUrl(string $callbackUrl)
             self::NOTIFICATION_CATEGORY_MUTATION
         );
 
-        UserPerson::update(
-            null, /* firstName */
-            null, /* middleName */
-            null, /* lastName */
-            null, /* publicNickName */
-            null, /* addressMain */
-            null, /* addressPostal */
-            null, /* avatarUuid */
-            null, /* taxResident */
-            null, /* documentType */
-            null, /* documentNumber */
-            null, /* documentCountryOfIssuance */
-            null, /* documentFrontAttachmentId */
-            null, /* documentBackAttachmentId */
-            null, /* dataOfBirth */
-            null, /* placeOfBirth */
-            null, /* countryOfBirth */
-            null, /* nationality */
-            null, /* language */
-            null, /* region */
-            null, /* gender */
-            null, /* status */
-            null, /* subStatus */
-            null, /* legalGuardianAlias */
-            null, /* sessionTimeout */
-            null, /* CardIds */
-            null, /* cardLimits */
-            null, /* dailyLimitWithoutConfirmationLogin */
-            $allUpdatedNotificationFilter
-        );
+        $userContext = BunqContext::getUserContext();
+        try {
```

Comment:
Please replace this try catch with an if statement that uses: https://github.com/bunq/sdk_php/blob/e9511e1c158a2c8d768d167bc05a66d7d88ea5d5/src/Context/UserContext.php#L120-L134

and then you can get the correct user without need to do `instanceof`

Reply:
That would indeed be nicer to have, but the return types aren't nullable, so you'll run into errors.

Reply:
nulls are bad anyway.

But i still do not see how `but the return types aren't nullable` will cause problems ?

```
if BunqContext::getUserContext()->isOnlyUserPersonSet()
user = getUserPerson()
update user person
else if  BunqContext::getUserContext()->isOnlyUserCompanySet()
user = getUserCompany
update user company 
else 
throw exception
```

Reply:
Thats the whole reason i've added thos bool methods :)

Reply:
Ahh, great! Now I see. Thanks! Just started using this library today and couldn't find it at first, because I'm not familiar enough.


> But i still do not see how `but the return types aren't nullable` will cause problems ?

Calling `$userContext->getUserCompany()` w/o a `UserCompany` set throws a `TypeError`. Is that wanted behavior?

What is the right exception to throw? `BunqException` or is there a specific one for this case? Again, please keep in mind I'm not familiar enough (yet) :slightly_smiling_face: